### DISCUSSION TITLE
AuthSessionMissingError: Auth session missing! when getUser() in an API route NEXT.JS.

### DISCUSSION THREAD

**Discussion by denull0** [supabase/supabase#28914] [Questions] [answered]
I am wondering if anyone else is having this problem. Supabase works in server actions, browser, middleware (able to fetched the current user there) but when i try to getUser() in an API Post route i get Auth session missing.

Thanks!

https://gist.github.com/denull0/ccf909d78941762ce3cc7a142d4e929a

**Comment by silentworks** [comment]
It's best to create a reproducible example repo with the issue. I've not had this issue before personally.

**Reply by kiikoh** [reply]
I had this the other day because I had my callback configured wrong. I was using a page.tsx instead of a route.ts.

**Comment by denull0** [comment, marked as answer]
My workaround is to pass cookies manually with the api call

```
      const cookiesStore = cookies();

      const metadataResponse = await fetch(
        `${process.env.NEXT_PUBLIC_URL}/api/insert-photo`,
        {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
            cookie: cookiesStore, // Pass the cookies manually
          },
        }
      );
```

**Reply by silentworks** [reply]
That is how fetch requests work. This is documented in the troubleshooting section https://github.com/orgs/supabase/discussions/15859

**Reply by denull0** [reply]
Omg all the debugging I had to do... Maybe it's not clear enough in their setup docs. I had no idea.

**Reply by james-william-r** [reply]
Damn. Yeah this really needs to be highlighted better in the docs. With the new SSR package, it kinda makes it sound/seem like this is already taken care off somehow. 😅
